crates/pop-cli/src/commands/call/contract.rs

```diff
@@ -119,26 +119,36 @@ impl CallContractCommand {
 	}
 
 	/// Checks if the contract has been built; if not, builds it.
+	/// If the path is a metadata file, skips the build process
 	async fn ensure_contract_built(&self, cli: &mut impl Cli) -> Result<()> {
-		// Check if build exists in the specified "Contract build directory"
-		if !has_contract_been_built(self.path.as_deref()) {
-			// Build the contract in release mode
-			cli.warning("NOTE: contract has not yet been built.")?;
-			let spinner = spinner();
-			spinner.start("Building contract in RELEASE mode...");
-			let result = match build_smart_contract(self.path.as_deref(), true, Verbosity::Quiet) {
-				Ok(result) => result,
-				Err(e) => {
-					return Err(anyhow!(format!(
+		if let Some(path) = self.path.as_deref() {
```

Comment:
What happens if the path is not specified - surely we should default to the current path and check as appropriate?

Something like `let path = self.path.unwrap_or(Path::new("./"))`

Reply:
I see that configure sets this - perhaps we should use an `.expect()` instead then?

Reply:
It makes sense to move the call to `ensure_contract_built` after the user has input the path.